From 034b4c38bbb0208d63e5409c3575774949854153 Mon Sep 17 00:00:00 2001
From: Zhang Le <r0bertz@gentoo.org>
Date: Thu, 12 Mar 2009 18:00:50 +0800
Subject: [PATCH 035/117] MIPS: compat: Move TIF_32BIT definition.

commit 034b4c38bbb0208d63e5409c3575774949854153 from linux-mips

This will prevent a build error after the merge with v2.6.27.20.

Signed-off-by: Zhang Le <r0bertz@gentoo.org>
Signed-off-by: Ralf Baechle <ralf@linux-mips.org>
(cherry picked from commit 85f3112271a8b4b4d987335f6f28652dd432542f)
---
 include/asm-mips/seccomp.h     |    4 ----
 include/asm-mips/thread_info.h |    6 ++++++
 2 files changed, 6 insertions(+), 4 deletions(-)

diff --git a/include/asm-mips/seccomp.h b/include/asm-mips/seccomp.h
index 36ed440..9e82daf 100644
--- a/include/asm-mips/seccomp.h
+++ b/include/asm-mips/seccomp.h
@@ -16,8 +16,6 @@
  */
 #ifdef CONFIG_MIPS32_O32
 
-#define TIF_32BIT TIF_32BIT_REGS
-
 #define __NR_seccomp_read_32		4003
 #define __NR_seccomp_write_32		4004
 #define __NR_seccomp_exit_32		4001
@@ -25,8 +23,6 @@
 
 #elif defined(CONFIG_MIPS32_N32)
 
-#define TIF_32BIT _TIF_32BIT_ADDR
-
 #define __NR_seccomp_read_32		6000
 #define __NR_seccomp_write_32		6001
 #define __NR_seccomp_exit_32		6058
diff --git a/include/asm-mips/thread_info.h b/include/asm-mips/thread_info.h
index bb30606..85c9a2d 100644
--- a/include/asm-mips/thread_info.h
+++ b/include/asm-mips/thread_info.h
@@ -126,6 +126,12 @@ register struct thread_info *__current_thread_info __asm__("$28");
 #define TIF_FPUBOUND		24	/* thread bound to FPU-full CPU set */
 #define TIF_SYSCALL_TRACE	31	/* syscall trace active */
 
+#ifdef CONFIG_MIPS32_O32
+#define TIF_32BIT TIF_32BIT_REGS
+#elif defined(CONFIG_MIPS32_N32)
+#define TIF_32BIT _TIF_32BIT_ADDR
+#endif /* CONFIG_MIPS32_O32 */
+
 #define _TIF_SYSCALL_TRACE	(1<<TIF_SYSCALL_TRACE)
 #define _TIF_SIGPENDING		(1<<TIF_SIGPENDING)
 #define _TIF_NEED_RESCHED	(1<<TIF_NEED_RESCHED)
-- 
1.7.0

